src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -271,14 +273,25 @@ public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
             .getVenueService()
             .getEpimsIdForVenue(caseData.getProcessingVenue())
             .orElse(null);
+        ObjectMapper mapper = new ObjectMapper();
+        List<String> epimsIdList = new ArrayList<>();
+        Map<String,List<String>> multipleHearingLocations = mapper.readValue(ResourceLoader.loadJson("multipleHearingLocations.json"), new TypeReference<Map<String, List<String>>>(){});
 
-        HearingLocation hearingLocation = new HearingLocation();
-        hearingLocation.setLocationId(epimsId);
-        hearingLocation.setLocationType(COURT);
+        List<List<String>> returnList = multipleHearingLocations.values().stream()
```

Comment:
We ought to be able to simplify this: if you key the map with the primary venue's Epims ID you could simply perform a get by the Epims ID then add all. @ss-evoco ?

Reply:
Agreed that this method can probably be simplified. Consider the following instead:
```
        return multipleHearingLocations.values().stream()
                .filter(listValues ->  listValues.contains(epimsId))
                .findFirst()
                .orElseGet(() -> Collections.singletonList(epimsId))
                
                .stream().map(epims -> HearingLocation.builder().locationId(epims).locationType(LocationType.COURT).build())
                .collect(Collectors.toCollection(ArrayList::new));
```
Provided some tests are conducted, I suspect it will be equivalent.

Whether this is more readable, on the other hand, can be debated.

Reply:
> We ought to be able to simplify this: if you key the map with the primary venue's Epims ID you could simply perform a get by the Epims ID then add all. @ss-evoco ?

This solution would be best if we assume that primary venue's Epims ID is will always be returned by the venue service. In case the venue service lookup returns the epims Id in the value? We would have to iterate both key entries as well as value entries, hence used a map with 3entries, so can can match either of the epims id in single iteration

Reply:
> Agreed that this method can probably be simplified. Consider the following instead:
> 
> ```
>         return multipleHearingLocations.values().stream()
>                 .filter(listValues ->  listValues.contains(epimsId))
>                 .findFirst()
>                 .orElseGet(() -> Collections.singletonList(epimsId))
>                 
>                 .stream().map(epims -> HearingLocation.builder().locationId(epims).locationType(LocationType.COURT).build())
>                 .collect(Collectors.toCollection(ArrayList::new));
> ```
> 
> Provided some tests are conducted, I suspect it will be equivalent.
> 
> Whether this is more readable, on the other hand, can be debated.

Beautifully written :)
Implemented and tested.

Reply:
@ss-evoco can we guarantee the primary will be returned?

Reply:
@dattipoeHMCTS @amittal89  No, we cannot assume the primary EPIMS will always come through (it could be any).